packages/cli/src/prebuild.ts

```diff
@@ -638,6 +638,33 @@ ${utilsExport}
     `
   );
 
+  const redirects = siteData.build.pages?.redirects;
+  if (redirects !== undefined && redirects.length > 0) {
+    spinner.text = "Generating redirects";
+
+    for (const redirect of redirects) {
+      const redirectPagePath = generateRemixRoute(redirect.old);
+      const redirectFileName = `${redirectPagePath}.ts`;
+
+      const content = `import { type LoaderArgs } from "@remix-run/server-runtime";
+import { redirect } from "@remix-run/server-runtime";
+import { matchPathFromRedirects } from "~/redirects";
+
+export const loader = (arg: LoaderArgs) => {
+  const redirectPath = matchPathFromRedirects(arg.request.url, ${JSON.stringify(
+    redirects
+  )});
```

Comment:
Why match at runtime if we already know new redirect here at build time?

Reply:
Thats a good catch, that got me thinking. We are not supporting dynamic destination at the moment. So, this logic can happen at cli time. And then just add the result string in the `redirect`. Will make the necessary changes 👍